cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -857,4 +873,41 @@ parquet_metadata read_parquet_metadata(host_span<std::unique_ptr<datasource> con
                           metadata.get_rowgroup_metadata()};
 }
 
+void reader::impl::update_output_nullmasks_for_pruned_pages(cudf::host_span<bool const> page_mask)
```

Comment:
@pmattione-nvidia added the function to update output buffer nullmasks right here.

Reply:
I am still kind of biased towards removing this function from the main reader's path since we are passing a `cudf::host_span<bool const> page_mask` containing all `true`s anyway from `decode_page_data`. This function can stay in the experimental path for now until its time comes. What do you guys think @pmattione-nvidia  @vuule  @nvdbaranec

Reply:
I would agree. Also, I think this would be worth profiling for a case with many pages.  If we had, say, 10000 pages, this is going to result in a _lot_ of kernel calls, which is usually pretty bad.  If we had a lot of pages and a lot of nesting (some customers can have hundreds of nested columns).   This seems like it is going to be a good candidate for a custom kernel.